cellpack/autopack/Ingredient.py

```diff
@@ -3241,86 +3341,81 @@ def checkSphCollisions(
                         )
-                        if not collision and level > 0:
-                            self.log.info("returning regular collision level")
-                            return collision, insidePoints, newDistPoints
-                        self.log.info("returning regular collision")
-                        return collision, insidePoints, newDistPoints
                     else:
-                        self.log.info("grid point already occupied %d", distance[pt])
+                        self.log.info("grid point already occupied %f", distance[pt])
                         return True, {}, {}
-                        # FIXME DEBUG INFO
-                        # dist = distA[pti]
-                        # d = dist - radc
-                        # if d + distance[pt] < histoVol.maxColl:
-                        #     histoVol.maxColl = d + distance[pt]
-                        #     # print("in collision histovol.maxColl if")
-                        # return True, insidePoints, newDistPoints
-                if pt in insidePoints:  # don't need to update any distances
+
+                if not at_max_level:
+                    # we don't want to calculate new distances if we are not
+                    # at the highest geo
+                    # but getting here means there was no collision detected
+                    # so the loop can continue
                     continue
 
                 signed_distance_to_sphere_surface = (
                     distance_to_packing_location - radius_of_ing_being_packed
                 )
+
                 if (
-                    signed_distance_to_sphere_surface <= 0 or signed_distance_to_sphere_surface <= 12.5
+                    signed_distance_to_sphere_surface <= 0
                 ):  # point is inside dropped sphere
                     if (
                         histoVol.grid.gridPtId[pt] != self.compNum and self.compNum <= 0
                     ):  # did this jitter outside of it's compartment
                         # in wrong compartment, reject this packing position
-                        # NOTE: Do we want to update distances if we're not packing at this location
-                        # MEGAN: I think we should return `True, {}, {}`
-                        return True, insidePoints, newDistPoints
+                        self.log.warning("checked pt that is not in container")
+                        return True, {}, {}
                     if pt in insidePoints:
-                        # NOTE: Don't we want the number closer to zero? Not the most negative?
-                        if signed_distance_to_sphere_surface > 0:
-                            signed_distance_to_sphere_surface = - signed_distance_to_sphere_surface
-                        if signed_distance_to_sphere_surface < insidePoints[pt]:
+                        if abs(signed_distance_to_sphere_surface) < abs(
+                            insidePoints[pt]
```

Comment:
This check seemed incorrect to me before, since they are both negative numbers and we want the one that tells us the closes surface, not the most negative number. I uses `abs` just so that it's consistent with the `min` check for distance below.